### ISSUE TITLE
Конфликты при старой информации в окнах

### ISSUE THREAD

**Issue by alexshpilkin** [fedor57/alumni-auth#7]
Ситуация:

> У меня введено два кода, помимо активного, — *A* и *B*, так что `session['codes']` = `['A', 'B']`. У меня открыто две вкладки, и я хоту отключить *A*.
>
> В первой вкладке я вижу включённый *A*, нажимаю «отключить»; откключается *A*. Проходит немного времени, я забываю всю эту историю.
>
> Натыкаюсь на вторую вкладку, вижу там включённый *A*, считаю, что мне показалось, и снова нажимаю «отключить»; **отключается _B_**.

Это можно решить, отключая код по коду, а не по его индексу в `session['codes']`, но тогда в коде странице будет написан код в полном виде, что не айс. Передавать номер и обфусцированный (`safe_form`) вид кода для проверки?

**Comment by fedor57** [comment]
Да, теоретически такая проблема есть, но, скорее всего, с ней никто не столкнется. 

Чтобы быть спокойней, предлагаю генерировать в качестве локального id  в сессии большое рандомное число, а не счетчик. Можно в него добавить время добавления в сессию.

**Comment by alexshpilkin** [comment]
О, действительно. Но рандом не обязательно: можно сделать именно что монотонный *счётчик* в сессии, а не индекс, как сейчас. Получится даже точно, а не вероятностно.

(Приятно, что при этом можно сделать все подтверждённые внутри сессии коды равноправными, что мне и так хочется сделать во избежание распухших голов у пользователей.)

**Comment by fedor57** [comment]
Нам нужно, чтобы при протирании сессии счетчик не перекрылся с другой сессией. Получается, что тогда в Application. Но глобальный счетчик выдает кол-во выписанных кодов, не хотелось бы лишнее показывать без явного решения.

Коды не совсем равноправные, должна быть коллекций, плюс - текущий в сессии, это важно. Отключать можно только не текущий, ибо это позволит убедиться в том, что пользователь будет знать хотя бы один неотключенный код. В урле я бы оставил /  для текущего.

**Comment by alexshpilkin** [comment]
При полном протирании сессии протухнет CSRF-токен в форме на старой странице, так что глобальный счётчик не нужен.

[Для протокола — обсуждение равноправия кодов было в слэке]

**Comment by fedor57** [comment]
Мы тут что-то рефакторили, проблемы ушли?
